### PR TITLE
Support `--user` and `--repo` argument for `config list` command

### LINKED ISSUE #2827: Support `--repo`, `--user`, and other target specifiers

### PR DESCRIPTION
**Pull request by daehyeok**
Fix #2827
# Example

user level configuration file(`"$HOME/Library/Application Support/jj/config.toml"`)
```toml
[user]
name = "Daehyeok Mun"
email = "daehyeok@gmail.com"

[ui]
pager = "bat -l toml --plain"
editor = "emacs -nw -q"
```

repo level configuration file(`.jj/repo/config.toml`)
```toml
[user]
email = "daehyeok@google.com"
```
example outputs.

```bash
#Release binary(v0.13.0)
 ~/workspace/jj ❯ jj config list

ui.editor="emacs -nw -q"
ui.pager="bat -l toml --plain"
user.name="Daehyeok Mun"
user.email="daehyeok@google.com"

#jj config list without any argument
#Doens't change anything.
 ~/workspace/jj ❯ ./target/debug/jj config list

ui.editor="emacs -nw -q"
ui.pager="bat -l toml --plain"
user.name="Daehyeok Mun"
user.email="daehyeok@google.com"

#print config options which defined in user level
#Doens't include overwriten value(user.email in this case)
 ~/workspace/jj ❯ ./target/debug/jj config list --user

ui.editor="emacs -nw -q"
ui.pager="bat -l toml --plain"
user.name="Daehyeok Mun"

#print config options which defined in repo level
 ~/workspace/jj ❯ ./target/debug/jj config list --repo

user.email="daehyeok@google.com"
```

# Checklist

If applicable:
- [X] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/commands/config.rs
Comment (martinvonz) [inline cli/src/commands/config.rs:95]: Since these values are mutually exclusive here (which seems reasonable), we should also tell clap that they are mutually exclusive so the user will get an error if they try `jj config list --user --repo`. You should be able to add `conflicts_with = "user"` in the attribute on `repo`.

**Review by martinvonz — COMMENTED** [review]
Thanks. Could you add a test or two as well?

**Comment by yuja** [comment]
If we add support for something like `includeIf` #616, should we consider these scoped configs as user's or repo's? It's not an immediate problem, but might be better to think about that.

**Comment by martinvonz** [comment]
> If we add support for something like `includeIf` #616, should we consider these scoped configs as user's or repo's? It's not an immediate problem, but might be better to think about that.

Good question. A related question is how we make `jj config list --show-origin` print such entries. The simplest solution would be to just print the full path there. And the simplest solution for `--repo` and `--user` would be to just include entries where we print the repo or user path (i.e. not includes). Maybe it would be confusing if `jj config list --user` did not include entries from a file included from the user config. `git config` has these flags:

```
       --show-origin
           Augment the output of all queried config options with the origin type (file, standard input, blob, command line) and the actual
           origin (config file path, ref, or blob id if applicable).
...
       --[no-]includes
           Respect include.*  directives in config files when looking up values. Defaults to off when a specific file is given (e.g.,
           using --file, --global, etc) and on when searching all config files.
```

**Comment by yuja** [comment]
> A related question is how we make `jj config list --show-origin` print such entries. The simplest solution would be to just print the full path there. And the simplest solution for `--repo` and `--user` would be to just include entries where we print the repo or user path (i.e. not includes). Maybe it would be confusing if `jj config list --user` did not include entries from a file included from the user config. `git config` has these flags:

I see. So in any case, `--repo`/`--user` applies to the physical location (not to the match condition of the `includeIf`.) That makes sense.

FWIW, I think it would be nice if `includeIf` could be inlined (so the project-specific values can be stored in one file.) It might add another complexity to the `config list` command, but the same rule can apply. If the value comes from ~/.jjconfig.toml, it will be displayed by `--user`.

**Comment by martinvonz** [comment]
> FWIW, I think it would be nice if `includeIf` could be inlined (so the project-specific values can be stored in one file.)

Yes, that's also a reasonable behavior for it.

@yuja: To clarify, you're not expecting any changes to this PR for that, right?

**Comment by yuja** [comment]
> @yuja: To clarify, you're not expecting any changes to this PR for that, right?

No, no change required. I haven't reviewed the code, but adding `--repo`/`--user` now makes sense to me.

File: cli/src/commands/config.rs
Comment (yuja) [inline cli/src/commands/config.rs:184]: nit: perhaps, --user/--repo would conflict with --include-defaults. They all apply to the `source`.

**Review by yuja — APPROVED** [review]

**Comment by martinvonz** [comment]
LGTM. Feel free to merge. Generally, once you have approval and there doesn't seem to be anything else left to discuss, feel to merge.

**Pull request merged**
